### PR TITLE
feat: word-wrap editor

### PR DESCRIPTION
**Pull request by cristianoventura**
## Description

This PR enables word-wrap for the Monaco editor in Script Preview and the Content tab of a response. The state for both are persisted individually in the settings file.

## How to Test

- open Script Preview and toggle the word-wrap switch
- open the response content of a JSON request and toggle word-wrap
- re-open the app and check that the previous state was persisted

## Checklist

- [x] I have performed a self-review of my code.
- [] I have added tests for my changes.
- [x] I have run linter locally (`npm run lint`) and all checks pass.
- [x] I have run tests locally (`npm test`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

## Screenshots (if appropriate):

https://github.com/user-attachments/assets/3b93ffa2-ed78-458d-9ae9-108ab4c01f58

## Related PR(s)/Issue(s)

Resolves https://github.com/grafana/k6-studio/issues/237

### CODE REVIEW

File: src/hooks/useEditorWordWrapSetting.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { EditorSettings } from '@/types/settings'
+import { useState, useEffect } from 'react'
+import { useDebounce } from 'react-use'
+
+export function useEditorWordWrapSetting(key: keyof EditorSettings) {
```
Comment (cristianoventura) [inline src/hooks/useEditorWordWrapSetting.ts:5]: This hook can be used to handle each setting independently based on the `key` (`wordWrapScriptPreview` or `wordWrapResponseContent`)

**Review by e-fisher — COMMENTED** [review]
I feel like word wrap toggle should be a part of CodeEditor component, perhaps a Toolbar component inside, with option to control it through props. 

Also, I probably wouldn't go as far as having this option saved per every component, just drop it in local storage to remember my preferred view and apply in all places we show editor. It's only one click away to toggle, anyway. IMO, added complexity doesn't outweigh benefit added, but would love to hear what others think.

**Review by cristianoventura — COMMENTED** [review]
I've followed @e-fisher's suggestion to keep the state in local storage. This indeed minimizes complexity. I also implemented a `EditorToolbar` component to hold the word wrap switch.

image

cc @going-confetti

**Review by Llandy3d — APPROVED** [review]
Looks good! 🙌 

I wonder if it makes sense to have it on a smaller size on the right side of the bar UI wise because I assume it is a "set and forget" element 🤔

**Review by e-fisher — COMMENTED** [review]
Agree with Llandy on moving it to the right. Maybe not just make it smaller, but use an icon with a tooltip instead? 

Found a weird bug when toggle doesn't when viewing `raw` request response, here's a video:

https://github.com/user-attachments/assets/4ea1207a-df73-46ef-b1ea-81d4043f9535

Otherwise looks great!

**Comment by cristianoventura** [comment]
I've moved the word wrap switch to an IconButton aligned at the end of the toolbar.

image

@e-fisher I noticed that this only happens for JSON >= than 10000 characters. It could be related to internal performance so I'm still trying to figure it out.

File: src/components/Monaco/ReactMonacoEditor.tsx [hunk 2]
```diff
@@ -36,14 +35,32 @@ export function ReactMonacoEditor({
...
 }: ReactMonacoEditorProps) {
   const theme = useTheme()
+  const [editor, setEditor] = useState<monaco.editor.IStandaloneCodeEditor>()
   const [toolbarState, setToolbarState] = useState<ToolbarState>({
     wordWrap: 'off',
   })
 
+  // Monaco automatically applies word wrap if the content length of a line is >= 10000 characters
+  // In this case, we disable the word wrap button so the internal state is respected
+  const shouldEnableWordWrapButton = () => {
```
Comment (cristianoventura) [inline src/components/Monaco/ReactMonacoEditor.tsx:45]: @e-fisher This was an interesting find. If any line has a content length >= 10000 characters, Monaco overrides the `wordWrap` option. This is likely related to [stopRenderingLineAfter](https://microsoft.github.io/monaco-editor/typedoc/interfaces/editor.IEditorConstructionOptions.html#stopRenderingLineAfter) but changing this option doesn't seem to have any effect on this case specifically other than hiding the rest of the line content.

This was the solution I came up with by disabling the word wrap button and respecting Monaco's internal state. Please let me know what you think.

**Review by Llandy3d — DISMISSED** [review]

**Review by e-fisher — DISMISSED** [review]
Great find, and sorry for taking you down the rabbit hole! 😅 

I would wrap the line calculation in `useMemo` to avoid running on every re-render, but otherwise looks good!

**Review by Llandy3d — APPROVED** [review]

**Pull request merged**